addons/mail/static/src/components/message_list/message_list.xml

```diff
@@ -64,12 +64,11 @@
                         <Message
                             class="o_MessageList_item o_MessageList_message"
                             t-att-class="{
-                                'o-has-message-selection': props.selectedMessageLocalId,
+                                'o-has-message-selection': threadView and threadView.threadViewer and threadView.threadViewer.selectedMessage,
```

Comment:
Move this class to message itself next to where `o-selected` is, because the css rule we just discussed only makes sense if both class are next to each other, it will be less confusing too.